frontend/ui/screensaver.lua

```diff
@@ -550,10 +556,31 @@ function Screensaver:show()
             widget_settings.image = self.image
             widget_settings.image_disposable = true
         elseif self.image_file then
-            widget_settings.file = self.image_file
-            widget_settings.file_do_cache = false
+            if G_reader_settings:isTrue("screensaver_autorotate_images") then
+                -- We need to load the image here to determine whether to rotate
+                if util.getFileNameSuffix(self.image_file) == "svg" then
+                    widget_settings.image = RenderImage:renderSVGImageFile(self.image_file, nil, nil, 1)
+                else
+                    widget_settings.image = RenderImage:renderImageFile(self.image_file, false, nil, nil)
+                end
+                if not widget_settings.image then
+                    widget_settings.image = RenderImage:renderCheckerboard(Screen:getWidth(), Screen:getHeight(), Screen.bb:getType())
+                end
+                widget_settings.image_disposable = true
+            else
+                widget_settings.file = self.image_file
+                widget_settings.file_do_cache = false
+            end
             widget_settings.alpha = true
         end
+        if G_reader_settings:isTrue("screensaver_autorotate_images") then
+            local angle = rotation_mode == 3 and 180 or 0 -- match mode if possible
+            if (widget_settings.image:getWidth() < widget_settings.image:getHeight())
+                ~= (widget_settings.width < widget_settings.height) then
+                angle = angle + 90
+            end
+            widget_settings.rotation_angle = angle
```

Comment:
This could be honoring the `imageviewer_rotation_portrait_invert` & `imageviewer_rotation_landscape_invert` settings for consistency's sake ;).

Reply:
Well, landscape, really, as Portrait will follow the current rotation.